pkg/ffmpeg/stream.go

```diff
@@ -0,0 +1,199 @@
+package ffmpeg
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/stashapp/stash/pkg/logger"
+	"github.com/stashapp/stash/pkg/models"
+)
+
+type Stream struct {
+	Stdout  io.ReadCloser
+	Stderr  io.ReadCloser
+	Process *os.Process
+}
+
+type Codec struct {
+	codec     string
+	format    string
+	MimeType  string
+	extraArgs []string
+}
+
+var CodecH264 = Codec{
+	codec:    "libx264",
+	format:   "mp4",
+	MimeType: MimeMp4,
+	extraArgs: []string{
+		"-movflags", "frag_keyframe",
+		"-preset", "veryfast",
+		"-crf", "30",
+	},
+}
+
+var CodecVP9 = Codec{
+	codec:    "libvpx-vp9",
+	format:   "webm",
+	MimeType: MimeWebm,
+	extraArgs: []string{
+		"-deadline", "realtime",
+		"-cpu-used", "5",
+		"-row-mt", "1",
+		"-crf", "30",
+	},
+}
+
+var CodecVP8 = Codec{
+	codec:    "libvpx",
+	format:   "webm",
+	MimeType: MimeWebm,
+	extraArgs: []string{
+		"-deadline", "realtime",
+		"-cpu-used", "5",
+		"-crf", "30",
+	},
+}
+
+var CodecHEVC = Codec{
+	codec:    "libx265",
+	format:   "mp4",
+	MimeType: MimeMp4,
+	extraArgs: []string{
+		"-movflags", "frag_keyframe",
+		"-preset", "veryfast",
+		"-crf", "30",
+	},
+}
+
+type streamOptions struct {
+	probeResult      VideoFile
+	codec            Codec
+	startTime        string
+	maxTranscodeSize models.StreamingResolutionEnum
+	videoOnly        bool
+}
+
+func (s streamOptions) getStreamArgs() []string {
+	scale := calculateTranscodeScale(s.probeResult, s.maxTranscodeSize)
+
+	args := []string{
+		"-hide_banner",
+		"-v", "error",
+	}
+
+	if s.startTime != "" {
+		args = append(args, "-ss", s.startTime)
+	}
+
+	args = append(args,
+		"-i", s.probeResult.Path,
+	)
+
+	if s.videoOnly {
+		args = append(args, "-an")
+	}
+
+	args = append(args,
+		"-c:v", s.codec.codec,
+		"-vf", "scale="+scale,
+	)
+
+	if len(s.codec.extraArgs) > 0 {
+		args = append(args, s.codec.extraArgs...)
+	}
+
+	args = append(args,
+		// this is needed for 5-channel ac3 files
+		"-af", "channelmap=channel_layout=5.1",
```

Comment:
As mentioned in discord 
`"-af", "channelmap=channel_layout=5.1",`
makes ffmpeg error out in my case in linux
the usage of the option is wrong and even then i dont think it's recommended for non 5.1 files
i would opt for `"-ac", "2",` instead to downmix to stereo